Source/WebCore/dom/Element.h

```diff
@@ -617,6 +618,8 @@ class Element : public ContainerNode {
     void addToTopLayer();
     void removeFromTopLayer();
 
+    void markRendererDirtyAfterTopLayerChange(RenderElement* renderer, RenderBlock* containingBlockBeforeStyleResolution);
```

Comment:
I think this should just be a helper function somewhere (and not an Element API)

Reply:
Agreed that this is a helper function, i moved it to RenderElement, let me know if it is appropriate.